cmake/module/ProcessConfigurations.cmake

```diff
@@ -0,0 +1,126 @@
+# Copyright (c) 2023 The Bitcoin Core developers
```

Comment:
Is this entire file (excluding the printing) all just coping with `-DNDEBUG`? If so I'm wondering if there isn't an easier way...

Reply:
> Is this entire file (excluding the printing) all just coping with `-DNDEBUG`?

This file's goal (excluding the printing) is to adjust CMake's default platform-specific [flags](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_FLAGS_CONFIG_INIT.html) to our needs, particularly, to drop the `NDEBUG` preprocessor macro, lower optimization level from `-O3` to `-O2`, and so on.

> If so I'm wondering if there isn't an easier way...

I'll be happy with any suggestion to implement :)

Reply:
Alternatively, we can define our custom default flags, which would override CMake's internal logic. However, I'd prefer to avoid this approach.